data/src/main/java/org/apache/iceberg/data/IcebergGenerics.java

```diff
@@ -41,13 +42,17 @@ public static ScanBuilder read(Table table) {
 
   public static class ScanBuilder {
     private final Table table;
-    private Expression where = Expressions.alwaysTrue();
-    private List<String> columns = ImmutableList.of("*");
+    private TableScan tableScan;
+    private final Expression defaultWhere = Expressions.alwaysTrue();
+    private final List<String> defaultColumns = ImmutableList.of("*");
     private boolean reuseContainers = false;
-    private boolean caseSensitive = true;
+    private final boolean defaultCaseSensitive = true;
 
     public ScanBuilder(Table table) {
       this.table = table;
+      this.tableScan = table.newScan()
+          .select(this.defaultColumns)
+          .caseSensitive(this.defaultCaseSensitive);
```

Comment:
Instead of having defaults here as well as in the table scan, this should just create a new scan and rely on its defaults.